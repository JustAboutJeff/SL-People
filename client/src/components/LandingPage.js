import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'

import { ConnectedNavBar } from '../containers/ConnectedNavBar'
import { Intro } from './Intro'

const Page = styled.div`
  display: grid;
  grid-template 46px 1fr / 1fr;
  height: 100%;
  width: 100%;
`

const FlexBox = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

const PeopleLink = styled(Link)`
  align-items: center;
  background: #4a90e2;
  border-radius: 7px;
  width: 827px;
  color: white;
  display: flex;
  font-size: 25px;
  font-weight: bold;
  justify-content: center;
  margin-top: 15px;
  padding: 5px;
  min-width: 152px;
  overflow: hidden;
`

export const LandingPage = () => (
  <Page>
    <ConnectedNavBar />
    <FlexBox>
      <Intro />
      <PeopleLink to='/people'>People Page</PeopleLink>
    </FlexBox>
  </Page>
)
