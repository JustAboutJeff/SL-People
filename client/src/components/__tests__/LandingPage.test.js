import React, { isValidElement } from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { LandingPage } from '../LandingPage'

describe('LandingPage', () => {
  let element, entry, renderer
  beforeEach(() => {
    element = <LandingPage />
    renderer = new ShallowRenderer()
  })

  it('returns a valid element', () => {
    expect(isValidElement(element)).toBe(true)
  })

  it('renders', () => {
    expect(renderer.render(element)).toMatchSnapshot()
  })
})
