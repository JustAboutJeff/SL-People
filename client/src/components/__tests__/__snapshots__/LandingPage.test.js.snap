// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders 1`] = `
<styled.div>
  <Connect(NavBar) />
  <styled.div>
    <Intro />
    <Styled(Link)
      to="/people"
    >
      People Page
    </Styled(Link)>
  </styled.div>
</styled.div>
`;
